File: Algorithm_problem_solving/Programers/Cording_Test/Thanksgiving_Traffic/Gym_suit.py
def solution(n, lost, reserve):
    # 위에서부터 순서대로 체육복이 없는 사람의 수, 각 사람이 가지고 있는 체육복의 수
    no_suit = 0
    gym_suit = [1] * (n+2)
    for i in reserve:
        gym_suit[i] += 1
    for i in lost:
        gym_suit[i] -= 1
    # 만약 체육복이 없다면 일단, 이미 검사한 사람이 체육복의 여분이 있다면 필린다.
    for i in range(1,n+1):
        if(gym_suit[i]==0):
            if(gym_suit[i-1]>1):
                gym_suit[i-1] -= 1
                gym_suit[i] = 1
            elif(gym_suit[i+1]>1):
                gym_suit[i+1] -= 1
                gym_suit[i] = 1
            else:
                no_suit+=1
    return n-no_suit

if __name__ == "__main__":
    n =	1
    lost = []
    reserve = []
    print(solution(n, lost, reserve))

File: Algorithm_problem_solving/Programers/Brute_Force/Gym_suit/Gym_suit.js
function solution(n, lost, reserve) {
    var answer = 0;
    var gym_suit = new Array(n + 2).fill(1);

    for (var i = 0; i < lost.length; i++) {
        gym_suit[lost[i]] = 0
    }
    for (var i = 0; i < reserve.length; i++) {
        gym_suit[reserve[i]] += 1
    }

    for (var i = 1; i < n + 1; i++) {
        if (gym_suit[i] < 1) {
            if (gym_suit[i - 1] > 1) {
                gym_suit[i - 1]--;
                gym_suit[i]++;
            }
            else if (gym_suit[i + 1] > 1) {
                gym_suit[i + 1]--;
                gym_suit[i]++;
            }
            else {
                answer++;
            }
        }
    }
    return n - answer;
}